src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -373,17 +373,18 @@ public function getCombinationIdsAction(int $productId, ProductCombinationFilter
     /**
      * @AdminSecurity("is_granted('delete', request.get('_legacy_controller'))")
      *
+     * @param Request $request
      * @param int $combinationId
      *
      * @return JsonResponse
      */
-    public function deleteAction(int $combinationId): JsonResponse
+    public function deleteAction(Request $request, int $combinationId): JsonResponse
```

Comment:
As we discussed maybe you can use the same "convention" I used for product list:
- in the routing add an optional `/{shopId}` at the end of the path
- you must define default values for shopId as null in the routing
- in the controller you can then rely on a `?int shopId` parameter
- when shopId is empty it means the constraint is allShops

It means the shopId must now be indicated explicitly in the route parameters, so you'll need to adapt `admin_products_combinations_delete_combination` and provide it the shopId (which will have to be passed to the component as a data attribute). But relying on an explicit parameter will make the controller more stable as this way we avoid relying on this ShopContext which is a source of error in so many ways